app/domain/operations/generate_saml_response.rb

```diff
@@ -36,8 +36,8 @@ def create_payment_transaction(hbx_enrollment, source)
       payment ? Success(payment) : Failure("Issue with Payment transcation")
     end
 
-    def init_saml_generator(payment_transaction, hbx_enrollment)
-      saml_obj = OneLogin::RubySaml::SamlGenerator.new(payment_transaction.payment_transaction_id, hbx_enrollment)
+    def init_saml_generator(payment_transaction, hbx_enrollment, pay_now_key)
```

Comment:
might want to clarify which key? pay_now_carrier_key, perhaps? not critical just a thought. At a glance i wasnt sure what is being passed around

Reply:
Thanks for commenting on this.  Got me to reconsider my previous refactor and also the existing Pay Now code, which led me to the realization that we don't need to pass the key around at all.  New approach requires fewer changes and should be more readable.  Let me know what you think!